aten/src/ATen/xpu/XPUGeneratorImpl.cpp

```diff
@@ -0,0 +1,176 @@
+#include <ATen/Utils.h>
+#include <ATen/xpu/XPUGeneratorImpl.h>
+#include <c10/core/StreamGuard.h>
+#include <c10/util/CallOnce.h>
+#include <c10/xpu/XPUFunctions.h>
+
+namespace at {
+namespace xpu::detail {
+namespace {
+
+/*
+ * Currently, there is one generator pool containing XPU generator per device.
+ * Each generator is lazily initialized when the first time generator is
+ * requested for a device.
+ */
+c10::once_flag init_flag;
+DeviceIndex num_gpus = -1;
+std::deque<c10::once_flag> xpu_gens_init_flag;
```

Comment:
Why not a vector like below?

Reply:
change to `std::vector<c10::once_flag> xpu_gens_init_flag;`
Here `std::vector` is more efficient than `std::deque`.

Reply:
Sorry @albanD, looks like `std::vector<c10::once_flag>` doesn't support `resize` method because `once_flag` lacks a copy constructor. see https://github.com/pytorch/pytorch/blob/685d862c450238683f07de54299c18ea16dfe086/c10/util/CallOnce.h#L37
I have also tried on [godbolt](https://godbolt.org/z/6q9G6n5n4). So, here I change back to `std::dequeu`.
I think I missed this error before because I didn't save my code change. Sorry for this.

Reply:
@albanD  Could you help review again?

Reply:
Ho interesting. Sounds good!